Upgrade versions

Diaphteiros opened the pull request:
**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       improvement|noteworthy|action
- target_group:   user|operator|developer
-->
```improvement operator
Upgrade 'virtual' kube-apiserver to `v1.16.6`
```
```improvement operator
Upgrade kubernetes versions to the latest patch versions. From now on, for each minor version, only one patch version will be supported (unless required otherwise in special cases).
```
```improvement operator
Upgrade Gardener extension 'provider-aws' to `v1.3.3`
```

Diaphteiros committed:
Upgrade virtual kube-apiserver to 1.15.10

Diaphteiros committed:
Upgrade kubernetes versions and keep only latest patch version for each minor version

Diaphteiros committed:
Upgrade Gardener extension provider-aws to v1.3.3

Diaphteiros commented:
/test --kubernetes-version=1.17.2

vpnachev commented:
The virtual kube-apiserver cannot be upgraded to v1.16, ref https://github.com/gardener/gardener/issues/1605

ialidzhikov commented:
> The virtual kube-apiserver cannot be upgraded to v1.16, ref [gardener/gardener#1605](https://github.com/gardener/gardener/issues/1605)

Yeah, nice catch.

Diaphteiros commented:
Oh, thanks for the info. I'll go with the latest 1.15 version, then.

Diaphteiros commented:
/test --kubernetes-version=1.17.2

Diaphteiros committed:
enhance/enable 'delete' plugin for Gardener resources

The 'delete' plugin will now
- annotate shoots and projects before deletion
- check for shoots before deleting a seed

Diaphteiros commented:
/test --host-cloudprovider=aws --kubernetes-version=1.17.2

vpnachev commented on components/gardencontent/profiles/manifests/manifest.yaml:
```diff
@@ -48,13 +48,10 @@ defaults:
-      - version: 1.15.6
+      - version: 1.17.2
+      - version: 1.16.6
+      - version: 1.15.9
```
Recently were released v1.17.3, v1.16.7 and v1.15.10
* https://github.com/kubernetes/kubernetes/releases/tag/v1.17.3
* https://github.com/kubernetes/kubernetes/releases/tag/v1.16.7
* https://github.com/kubernetes/kubernetes/releases/tag/v1.15.10

Diaphteiros replied:
Thanks for the info, I'll update the commit.
